ansible/roles/ocp-workload-edge-deployments/tasks/workload.yml

```diff
@@ -0,0 +1,133 @@
+  shell: "sed -i 's/YOUR_TOKEN_HERE/{{ tokenresult.stdout }}/g' /tmp/{{guid}}/deploy-containers-apb/playbooks/provision.yml"
+
+- name: Setting the OCP Registry Address in the APB configuration
+  shell: "sed -i 's/YOUR_OCP_ADDRESS_HERE/docker-registry-default.{{ ocp_apps_domain }}/g' /tmp/{{guid}}/deploy-containers-apb/playbooks/provision.yml"
+
+- name: Setting the Endpoint for AMQ connections (supposing a Single Node Installation) in the APB configuration
+  shell: "sed -i 's/YOUR_INFRA_ADDR/node1.{{guid}}.internal/g' /tmp/{{guid}}/deploy-containers-apb/playbooks/provision.yml"
+
+- name: Ensure that APB binary is installed
+  become: true
+  yum:
+    name: apb
+    state: present
+
+- name: Ready to prepare the APB build
+  shell: "cd /tmp/{{guid}}/deploy-containers-apb; apb prepare"
+
+- name: Create the APB build
+  shell: "cd /tmp/{{guid}}/deploy-containers-apb; oc new-build --name deploy-containers-apb --allow-missing-images -n openshift ."
+
+- name: Start the APB build and push process
+  shell: "cd /tmp/{{guid}}/deploy-containers-apb; oc start-build -n openshift --follow --from-dir . deploy-containers-apb"
+
+- name: Annotate the completed iot-development project as requested by user
+  shell: "oc annotate namespace iot-development openshift.io/requester={{ocp_username}} --overwrite"
+
+- name: Annotate the completed iot-testing project as requested by user
+  shell: "oc annotate namespace iot-testing openshift.io/requester={{ocp_username}} --overwrite"
+
+- name: Annotate the completed iot-hub project as requested by user
+  shell: "oc annotate namespace iot-hub openshift.io/requester={{ocp_username}} --overwrite"
+
+- name: Give user access to the completed iot-development project
+  shell: "oc policy add-role-to-user admin {{ocp_username}} -n iot-development"
+
+- name: Give user access to the completed iot-testing project
+  shell: "oc policy add-role-to-user admin {{ocp_username}} -n iot-testing"
+
+- name: Give user access to the completed iot-hub project
+  shell: "oc policy add-role-to-user admin {{ocp_username}} -n iot-hub"
+
+# Fixing fis-karaf issue with registry.redhat.io
+- name: Backing up the fis-karaf ImageStream
+  shell: "oc get --export -n openshift is fis-karaf-openshift -o yaml | sed 's/registry.redhat.io/registry.access.redhat.com/g' > /tmp/{{guid}}/fis-karaf-openshift-is-registry.yaml"
+
+- name: Deleting the fis-karaf ImageStream
+  shell: "oc delete -n openshift is fis-karaf-openshift"
+
+- name: Creating the fis-karaf ImageStream
+  shell: "oc create -n openshift -f /tmp/{{guid}}/fis-karaf-openshift-is-registry.yaml"
+
+# Starting the configuration for the External Machine. We suppose that the External machine is the bastion where we're launching this ansible playbook.
+- name: Adding SSH public key to the authorized host for the machine
+  become: true
+  shell: "cat /tmp/{{guid}}/deploy-containers-apb/id_rsa.pub >> /root/.ssh/authorized_keys; chmod 700 /root/.ssh/authorized_keys"
```

Comment:
you can use the `authorized_key` module https://docs.ansible.com/ansible/latest/modules/authorized_key_module.html